### PR TITLE
:bug: Fix link to v1alpha4 types

### PR DESCRIPTION
**Pull request by killianmuldoon**
Fixup a link to v1alpha4 go code in an older proposal.

/area documentations

### CODE REVIEW

File: docs/proposals/20181121-machine-api.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ performed in-place or via Node replacement.
 ## Proposal
 
 This proposal introduces a new API type: Machine. See the full definition in
-[machine_types.go](../../api/v1alpha4/machine_types.go).
+[machine_types.go](https://github.com/kubernetes-sigs/cluster-api/blob/release-0.4/api/v1alpha4/machine_types.go).
```
Comment (killianmuldoon) [inline docs/proposals/20181121-machine-api.md:44]: Makes sense to me to link to the same generation of types the proposal was talking about.

**Comment by sbueringer** [comment]
Thx!

**Review by chrischdi — COMMENTED** [review]
/lgtm

**Pull request merged**